src/powershell/Program.cs

```diff
@@ -19,7 +59,439 @@ public sealed class ManagedPSEntry
+                    }
+
+                    // Skip over argc, remember where exec_path is
+                    execPathPtr = IntPtr.Add(procargs, sizeof(int));
+
+                    // Skip over exec_path
+                    byte *argvPtr = (byte *)execPathPtr;
+                    while (*argvPtr != 0) { argvPtr++; }
+                    while (*argvPtr == 0) { argvPtr++; }
+
+                    // First char in argv[0]
+                    procNameFirstByte = *argvPtr;
+                }
+
+                if (!IsLogin(procNameFirstByte, args, out int loginArgIndex))
+                {
+                    return;
+                }
+
+                // Get the pwshPath from exec_path
+                pwshPath = Marshal.PtrToStringAnsi(execPathPtr);
+
+                // exec pwsh
+                ThrowIfFails("exec", ExecPwshLogin(args, loginArgIndex, pwshPath, isMacOS: true));
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(procargs);
+            }
+        }
+
+        /// <summary>
+        /// Checks args to see if -Login has been specified.
+        /// </summary>
+        /// <param name="procNameFirstByte">The first byte of the name of the currently running process.</param>
+        /// <param name="args">Arguments passed to the program.</param>
+        /// <param name="loginIndex">The arg index (in argv) where -Login was found.</param>
+        /// <returns></returns>
+        private static bool IsLogin(
+            byte procNameFirstByte,
+            string[] args,
+            out int loginIndex)
+        {
+            loginIndex = -1;
+
+            switch (procNameFirstByte)
+            {
+                // '+' signifies we have already done login check
+                case 0x2B:
+                    return false;
+
+                // '-' means this is a login shell
+                case 0x2D:
+                    return true;
+
+                // For any other char, we check for a login parameter
+            }
+
+            // Parameter comparison strings, stackalloc'd for performance
+            for (int i = 0; i < args.Length; i++)
+            {
+                string arg = args[i];
+
+                // Must look like '-<name>'
+                if (arg == null || arg.Length < 2 || arg[0] != '-')
+                { 
+                    continue;
+                }
+
+                // Check for "-Login" or some prefix thereof
+                if (IsParam(arg, "login", "LOGIN"))
+                {
+                    loginIndex = i;
+                    return true;
+                }
+
+                // After -File and -Command, all parameters are passed
+                // to the invoked file or command, so we can stop looking.
+                if (IsParam(arg, "file", "FILE")
+                    || IsParam(arg, "command", "COMMAND"))
```

Comment:
Shall we check for `-Version | -v` as well here? Flags specified after `-v` will be ignored.
Are there any other similar flags?

Reply:
Ah good catch!

Reply:
How about `-EncodedCommand` and `-Help`? Docs don't say it, but they also don't say `-EncodedCommand` is mutually exclusive with `-Command`. I tried `-?` with `-Version` and it prints both, but `-?` before `-Command` only shows the help...